### ISSUE TITLE
Send selected image sending the wrong image

### ISSUE THREAD

**Issue by Mozoloa** [sinano1107/sd-webui-auto-tweet#4]
https://github.com/sinano1107/sd-webui-auto-tweet/blob/fcf9355de60612e3b1494572fb72575b13f8d786/scripts/extension.py#L73-L79

This seems broken as long as the gallery is still open, here's why, and how to reproduce:
- Launch webui
- Generate 1 img and select it, opening the image in this view :
![image](https://github.com/sinano1107/sd-webui-auto-tweet/assets/20817233/d37e2ce2-68a9-47f4-949c-4fae5226193d)
- `selected_imgae` updates, so you can send this image
- Generate another one with the gallery still open in picture view
- `selected_imgae` has not been updated, sending will send the previous image
- Now try generating 2 or more (batch size or count) while having the gallery still open
- `selected_imgae` is still not updated
- Select another image in the gallery than the one that's currently displayed (usually the first one)
- NOW `selected_imgae` is correctly updated

So basically if the gallery is open then `selected_imgae` doesn't update with the new image, even if it's open and the only one.

Maybe watching `gr.SelectData` isn't the right way to go or maybe we need to look into something else as well at the same time ?
